packages/framework/src/validators/json-schema.validator.ts

```diff
@@ -20,7 +20,7 @@ export class JsonSchemaValidator implements Validator<JsonSchema> {
       // https://ajv.js.org/options.html#usedefaults
       useDefaults: true,
       // https://ajv.js.org/options.html#removeadditional
-      removeAdditional: 'all',
+      removeAdditional: 'failing',
```

Comment:
This was the important change to ensure that specifying `additionalProperties: true` on the schema is acted upon, maintaining the additional properties. Previously, all additional properties were being removed, ignoring this schema property.